schema_based_on_diagram.sql

```diff
@@ -0,0 +1,62 @@
+/* Database schema to keep the structure of entire database. */
+
+DROP TABLE IF EXISTS patients;
+CREATE TABLE patients (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  name VARCHAR(100),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+DROP TABLE IF EXISTS medical_histories;
+CREATE TABLE medical_histories (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR(100),
+  PRIMARY KEY(id),
+  FOREIGN KEY (patient_id) REFERENCES patients(id) ON UPDATE CASCADE ON DELETE CASCADE
+);
+
+DROP TABLE IF EXISTS invoices;
+CREATE TABLE invoices (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id) ON UPDATE CASCADE ON DELETE CASCADE
+);
+
+DROP TABLE IF EXISTS treatments;
+CREATE TABLE treatments (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  type VARCHAR(100),
+  name VARCHAR(100),
+  PRIMARY KEY(id)
+);
+
+DROP TABLE IF EXISTS invoice_items;
+CREATE TABLE invoice_items (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id) ON UPDATE CASCADE ON DELETE CASCADE,
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id) ON UPDATE CASCADE ON DELETE CASCADE
+);
+
+/* Joining tables */
+DROP TABLE IF EXISTS medical_treatment;
+CREATE TABLE medical_treatment (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  medical_history_id INT,
+  treatment_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id) ON UPDATE CASCADE ON DELETE CASCADE,
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id) ON UPDATE CASCADE ON DELETE CASCADE
+);
```

Comment:
Please create all FOREIGN KEYS indexes.

For example:

* CREATE INDEX ON medical_histories (patient_id);
* CREATE INDEX ON invoices (medical_history_id);
* CREATE INDEX ON invoice_items (invoice_id);

...and so forth